lib/tests/streamlit/date_input_test.py

```diff
@@ -52,3 +55,22 @@ def test_value_types(self, arg_value, proto_value):
         c = self.get_delta_from_queue().new_element.date_input
         self.assertEqual(c.label, "the label")
         self.assertEqual(c.default, proto_value)
+
+    @parameterized.expand(
+        [
+            (date(1961, 4, 12), "1951/04/12", "1971/04/12"),
+            (date(2020, 2, 29), "2010/02/28", "2030/02/28"),
+            # TODO: Find a way to mock date.today()
```

Comment:
I haven't been able to find a way to mock the date.today() function
A possible solution is to use the [freezgun]( https://github.com/spulec/freezegun) library.

Reply:
@vdonato I tried this option and several options with a modified patch path

Unfortunately, everything falls with errors

For example this:
```
def test_min_max_values_missing_value(self):
    with patch("streamlit.delta_generator.DeltaGenerator.date_input.date") as mock_date:
        mock_date.today.return_value = date(1999, 10, 8)
        mock_date.side_effect = lambda *args, **kw: date(*args, **kw)

        st.date_input("the label", [])
        c = self.get_delta_from_queue().new_element.date_input
        self.assertEqual(c.min, "1989/10/08")
        self.assertEqual(c.max, "2009/10/08")
``` 
fails with error
`AttributeError: <function TimeWidgetsMixin.date_input at 0x1526e48b0> does not have the attribute 'date'`

I think we will find a way to mock dates in tests. Or we will start using a library for this.

For now, I will leave a TODO comment, to come back to this later.